pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -108,26 +108,33 @@ class ModelCheckpoint(Callback):
         every_n_train_steps: Number of training steps between checkpoints.
             If ``every_n_train_steps == None or every_n_train_steps == 0``, we skip saving during training.
             To disable, set ``every_n_train_steps = 0``. This value must be ``None`` or non-negative.
-            This must be mutually exclusive with ``train_time_interval`` and ``every_n_val_epochs``.
+            This must be mutually exclusive with ``train_time_interval`` and ``every_n_epochs``.
         train_time_interval: Checkpoints are monitored at the specified time interval.
             For all practical purposes, this cannot be smaller than the amount
             of time it takes to process a single training batch. This is not
             guaranteed to execute at the exact time specified, but should be close.
-            This must be mutually exclusive with ``every_n_train_steps`` and ``every_n_val_epochs``.
-        every_n_val_epochs: Number of validation epochs between checkpoints.
-            If ``every_n_val_epochs == None or every_n_val_epochs == 0``, we skip saving on validation end.
-            To disable, set ``every_n_val_epochs = 0``. This value must be ``None`` or non-negative.
+            This must be mutually exclusive with ``every_n_train_steps`` and ``every_n_epochs``.
+        every_n_epochs: Number of epochs between checkpoints.
```

Comment:
```suggestion
        every_n_epochs: Number of training epochs between checkpoints.
```
not to be ambiguous, but I guess it is already clear we mean training epochs

Reply:
The rename in this PR is so that it can be used for validation epochs and training epochs.

So we cannot specify here

Reply:
@carmocca why is every_n_epochs along with a boolean flag to save on train epoch end preferable to supporting two flags like every_n_val_epochs and every_n_train_epochs?

The latter allows us to adjust the checkpoint frequency separately for training and validation. The mutual exclusion check is something we could address in the callback implementation